### PR TITLE
Add more GoalState

### PR DESCRIPTION
**Pull request by DavidLiu506**

### CODE REVIEW

File: kubernetes/services/dpm_manager.yaml [hunk 1]
```diff
@@ -62,8 +62,8 @@ data:
     group.topic.to.multicast.topic.map=multicast-topic1:group-topic1,group-topic3 multicast-topic2:group-topic2,group-topic4
 
 
-    arionGateway.enabled = false
-    arionMaster.server = 127.0.0.1
+    arionGateway.enabled = true
```
Comment (lfu-ps) [inline kubernetes/services/dpm_manager.yaml:65]: I assume we should leave it 'false' by default? To not break legacy E2E testing

File: services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/service/impl/ArionWingService.java [hunk 2]
```diff
@@ -55,37 +54,27 @@ public void buildArionGatewayState (NetworkConfiguration networkConfiguration, U
...
+            gatewayStateBuilder.getConfigurationBuilder().setGatewayType(Gateway.GatewayType.ARION);
+            gatewayStateBuilder.getConfigurationBuilder().getArionInfoBuilder().setPortInbandOperation(8300);
+            unicastGoalStateV2.getGoalStateBuilder().putGatewayStates(id, gatewayStateBuilder.build());
         }
     }
 
-    public String getArionGroup (int vni, String subnet) {
-        String key = String.valueOf(vni) + "-" + subnet;
-        Optional<SimpleNode> group = ring.locate(key);
+    public String getArionGroup (String subnetId) {
```
Comment (lfu-ps) [inline services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/service/impl/ArionWingService.java:68]: Why remove the param of vni? Is this API happening inside a vpc, so you don't need the vni? 
I assume subnets could be the same across different vpcs.

**Review by lfu-ps — CHANGES_REQUESTED** [review]

File: kubernetes/services/dpm_manager.yaml (hunk 1 above)
Comment (DavidLiu506) [reply kubernetes/services/dpm_manager.yaml:65]: @lfu-ps you are correct, thanks.

File: services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/service/impl/ArionWingService.java (hunk 2 above)
Comment (DavidLiu506) [reply services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/service/impl/ArionWingService.java:68]: subnetId is UUID

Comment (lfu-ps) [reply services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/service/impl/ArionWingService.java:68]: Make sense

**Review by lfu-ps — APPROVED** [review]

File: services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/service/impl/NeighborService.java [hunk 3]
```diff
@@ -113,8 +117,9 @@ public Neighbor.NeighborState buildNeighborState(NeighborEntry.NeighborType type
         fixedIpBuilder.setIpAddress(portHostInfo.getPortIp());
         fixedIpBuilder.setNeighborType(neighborType);
         if (arionGatwayEnabled) {
+            neighborConfigBuilder.setHostMacAddress(nodeInfoCache.getNodeInfoByNodeIp(portHostInfo.getHostIp()).get(0).getMacAddress());
```
Comment (xieus) [inline services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/service/impl/NeighborService.java:120]: Is it possible that nodeInfoCache.getNodeInfoByNodeIp(portHostInfo.getHostIp()) is empty?

File: kubernetes/services/dpm_manager.yaml [hunk 4]
```diff
@@ -63,7 +63,7 @@ data:
 
 
     arionGateway.enabled = false
-    arionMaster.server = 127.0.0.1
+    arionMaster.server = 10.213.43.71
```
[the remark this answers is not in the document]
Comment (DavidLiu506) [reply kubernetes/services/dpm_manager.yaml:66]: ok, thanks @xieus

File: services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/service/impl/NeighborService.java (hunk 3 above)
Comment (lfu-ps) [reply services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/service/impl/NeighborService.java]: Fixed

**Pull request merged**